Gem/Code/Source/Automation/ScriptReporter.cpp

```diff
@@ -419,255 +421,145 @@ namespace AtomSampleViewer
                         }
-                        if (localBaselineWarning)
+
+                        // Number of screenshot failures
+                        HighlightTextIf(scriptReport.m_screenshotErrorCount > 0, m_highlightSettings.m_highlightFailed);
+                        if (m_showAll || scriptReport.m_screenshotErrorCount > 0)
                         {
-                            headerSummary += "(Local Baseline Warning)";
+                            ImGui::Text("Screenshot Tests Failed: %u %s", scriptReport.m_screenshotErrorCount, SeeBelow(scriptReport.m_screenshotErrorCount).c_str());
                         }
 
-                        ImGuiTreeNodeFlags screenshotNodeFlag = FlagDefaultClosed;
-                        AZStd::string screenshotHeader = AZStd::string::format("%s %s %s", screenshotPassed ? "PASSED" : "FAILED", fileName.c_str(), headerSummary.c_str());
-
-                        HighlightTextFailedOrWarning(!screenshotPassed, localBaselineWarning);
-                        if (ImGui::TreeNodeEx(&screenshotResult, screenshotNodeFlag, "%s", screenshotHeader.c_str()))
+                        // Number of screenshot warnings
+                        HighlightTextIf(scriptReport.m_screenshotWarningCount > 0, m_highlightSettings.m_highlightWarning);
+                        if (m_showAll || (m_showWarnings && scriptReport.m_screenshotWarningCount > 0))
                         {
-                            ResetTextHighlight();
-
-                            ImGui::Text("Screenshot:        %s", screenshotResult.m_screenshotFilePath.c_str());
+                            ImGui::Text("Screenshot Warnings:     %u %s", scriptReport.m_screenshotWarningCount, SeeBelow(scriptReport.m_screenshotWarningCount).c_str());
+                        }
 
-                            ImGui::Spacing();
+                        ResetTextHighlight();
 
-                            HighlightTextIf(!screenshotPassed, m_highlightSettings.m_highlightFailed);
+                        for (ScreenshotTestInfo& screenshotResult : scriptReport.m_screenshotTests)
+                        {
+                            const bool screenshotPassed = screenshotResult.m_officialComparisonResult.m_resultCode == ImageComparisonResult::ResultCode::Pass;
+                            const bool localBaselineWarning = screenshotResult.m_localComparisonResult.m_resultCode != ImageComparisonResult::ResultCode::Pass;
 
-                            ImGui::Text("Official Baseline: %s", screenshotResult.m_officialBaselineScreenshotFilePath.c_str());
+                            AZStd::string fileName;
+                            AzFramework::StringFunc::Path::GetFullFileName(screenshotResult.m_screenshotFilePath.c_str(), fileName);
 
-                            // Official Baseline Result
-                            ImGui::Indent();
+                            AZStd::string headerSummary;
```

Comment:
In this case, I think using string stream would be a better choice, since it is appending and constructing string all the time.

Reply:
Thanks for looking! I've refactored to use stringstream.